src/databricks/labs/ucx/workspace_access/groups.py

```diff
@@ -86,40 +86,50 @@ def is_in_scope(self, name: str) -> bool:
     def __len__(self):
         return len(self._name_to_group)
 
-    def apply_group_permissions_experimental(self, ws: WorkspaceClient) -> bool:
+    def apply_to_renamed_groups(self, ws: WorkspaceClient) -> bool:
+        """(Production) Apply migration state to groups that have been renamed in the workspace."""
+        return self._apply_to_groups(ws, renamed=True)
+
+    def apply_to_groups_with_different_names(self, ws: WorkspaceClient) -> bool:
+        """(Integration Testing) Apply to groups that have different names in the workspace and account."""
+        return self._apply_to_groups(ws, renamed=False)
+
+    def _apply_to_groups(self, ws: WorkspaceClient, *, renamed: bool = False) -> bool:
```

Comment:
curios to know why we have "*" as second parameter here, maybe we can remove if it's not needed.